cli/flox/src/commands/mod.rs

```diff
@@ -886,34 +888,114 @@ impl UninitializedEnvironment {
+            UninitializedEnvironment::Remote(pointer) => &pointer.name,
+        }
+    }
+
+    /// Returns true if the environment is in the current directory
+    pub fn is_current_dir(&self) -> Result<bool> {
+        match self {
+            UninitializedEnvironment::DotFlox(DotFlox { path, .. }) => {
+                let current_dir = std::env::current_dir()?;
+                let is_current = current_dir.canonicalize()? == path.canonicalize()?;
+                Ok(is_current)
             },
+            UninitializedEnvironment::Remote(_) => Ok(false),
+        }
+    }
+
+    /// Returns the path to the environment if it isn't remote
+    #[allow(dead_code)]
+    pub fn path(&self) -> Option<&Path> {
+        match self {
+            UninitializedEnvironment::DotFlox(DotFlox { path, .. }) => Some(path),
+            UninitializedEnvironment::Remote(_) => None,
+        }
+    }
+
+    /// Returns true if the environment is a managed environment
+    pub fn is_managed(&self) -> bool {
+        match self {
+            UninitializedEnvironment::DotFlox(DotFlox { pointer, .. }) => pointer.owner().is_some(),
+            UninitializedEnvironment::Remote(_) => false,
         }
     }
+
+    /// Returns true if the environment is a path environment
+    #[allow(dead_code)]
+    pub fn is_path_env(&self) -> bool {
+        match self {
+            UninitializedEnvironment::DotFlox(DotFlox { pointer, .. }) => pointer.owner().is_none(),
+            UninitializedEnvironment::Remote(_) => false,
+        }
+    }
+
+    /// Returns true if the environment is a remote environment
+    pub fn is_remote(&self) -> bool {
+        match self {
+            UninitializedEnvironment::DotFlox(_) => false,
+            UninitializedEnvironment::Remote(_) => true,
+        }
+    }
+
+    /// The environment description when displayed in a prompt
+    pub fn prompt_description(&self) -> Result<String> {
+        if self.is_remote() {
+            Ok(format!(
+                "{}/{} (remote)",
+                self.owner().unwrap(),
+                self.name()
+            ))
+        } else if self.is_managed() {
+            Ok(format!("{}/{}", self.owner().unwrap(), self.name()))
+        } else {
+            Ok(format!("{}", self.name()))
+        }
+    }
+
+    /// The environment description when displayed in messages
+    pub fn message_description(&self) -> Result<String> {
+        if self.is_remote() {
+            Ok(format!(
+                "'{}/{}' (remote)",
+                self.owner().unwrap(),
+                self.name()
+            ))
+        } else if self.is_managed() {
+            Ok(format!("'{}/{}'", self.owner().unwrap(), self.name()))
+        } else if self
+            .is_current_dir()
+            .expect("couldn't read current directory")
+        {
+            Ok(String::from("in current directory"))
+        } else {
+            Ok(format!("'{}'", self.name()))
+        }
+    }
+}
```

Comment:
This does not need to return a Result, all branches either panic (which i think is ok --  perhaps we can convert not finding a current directory for a better description and not panic).